sources/packages/backend/apps/api/src/route-controllers/overaward/overaward.controller.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from "@nestjs/common";
+import { DisbursementOverawardService } from "@sims/services";
+import { DisbursementOveraward } from "@sims/sims-db";
+import { getUserFullName } from "../../utilities";
+import {
+  OverawardAPIOutDTO,
+  OverawardBalanceAPIOutDTO,
+} from "./models/overaward.dto";
+
+/**
+ * Overaward controller service.
+ */
+@Injectable()
+export class OverawardControllerService {
+  constructor(
+    private readonly disbursementOverawardService: DisbursementOverawardService,
+  ) {}
+
+  /**
+   * Get the overaward balance of a student.
+   * @param studentId student.
+   * @returns overaward balance for student.
+   */
+  async getOverawardBalance(
+    studentId?: number,
+  ): Promise<OverawardBalanceAPIOutDTO> {
+    const overawardBalance =
+      await this.disbursementOverawardService.getOverawardBalance([studentId]);
+
+    return { overawardBalanceValues: overawardBalance[studentId] };
+  }
+
+  /**
+   * Get all overawards which belong to a student.
+   * @param studentId student.
+   * @param includeAddedBy include added by.
+   * @returns overaward details of a student.
+   */
+  async getOverawardsByStudent(
+    studentId?: number,
+    includeAddedBy = false,
+  ): Promise<OverawardAPIOutDTO[]> {
+    const overawards =
+      await this.disbursementOverawardService.getOverawardsByStudent(studentId);
+    return overawards.map((overaward) =>
+      this.transformOverawards(overaward, includeAddedBy),
```

Comment:
Just as a suggestion, for such a small method, I would not mind having the content from `this.transformOverawards` in this method also.